### PR TITLE
wait_for_ready(): refactor

### PR DESCRIPTION
**Pull request by edsantiago**
add a common 'wait_ready' setting (default: 60s). Allow individual
subtest .ini files to override, and continue allowing overriding
in code itself.

Goal is to eliminate various 5- and 10-second timeouts: these are
causing sporadic and irreproducible test failures.

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

File: config_defaults/defaults.ini [hunk 1]
```diff
@@ -22,6 +22,9 @@ docker_options =
 #: Max runtime in seconds for any docker command (auto-converts to float)
 docker_timeout = 300.0
 
+#: Time (seconds) to wait for READY from container
```
Comment (cevich) [inline config_defaults/defaults.ini]: Small thing: Might be nice to future developers (stolen phrase) to mention this is not automatic, it's for use with ``dockercmd.wait_for_ready()``.

Comment (cevich) [reply config_defaults/defaults.ini]: How about:

```
#: Time (seconds) to wait for "READY" from the container, when using
#: ``dockercmd.AsyncDockerCmd.wait_for_ready()``
```

Comment (cevich) [reply config_defaults/defaults.ini]: Actually...in ``wait_for_ready()`` it has access to the test configuration.  Would it make sense if that method used ``self.subtest.config["wait_ready"]`` as the default value ``if timeout is None``?

Comment (cevich) [reply config_defaults/defaults.ini]: Yeah, looks like it.  Right now the default is ``self.timeout`` which IIRC is the 'docker_timeout' key...that's probably NOT what we want, using your new value would be much better IMHO.

Comment (edsantiago) [reply config_defaults/defaults.ini]: I'm confused. Isn't that exactly what this PR is doing?

Comment (cevich) [reply config_defaults/defaults.ini]: lol durrr....just looked at code.  I totally missed that change in dockercmd.

Comment (cevich) [reply config_defaults/defaults.ini]: How about:

```
#: Default timeout (seconds) to use, when calling
#: ``dockercmd.AsyncDockerCmd.wait_for_ready()`` with ``timeout=None``.
```

Comment (edsantiago) [reply config_defaults/defaults.ini]: Force-pushed with updated comment.

**Pull request merged**